core/src/test/java/org/kohsuke/stapler/DataBindingTest.java

```diff
@@ -328,6 +331,56 @@ public void testFieldInjection() {
         assertEquals(10,r.post);
     }
 
+    public static class BeanValidation {
+
+
+        @DataBoundSetter
+        @Trim @DefaultValue("test")
+        private String w;
+
+        @DataBoundSetter
+        @Positive @DefaultValue("1")
+        private int x;
+
+        @DataBoundSetter
+        @NotBlank
+        private String y;
+
+        @DataBoundSetter
+        private String z;
+
+        void assertValues() {
+            assertEquals("test",w);
+            assertEquals(1,x);
+            assertEquals("2",y);
+            assertEquals("3",z);
+        }
+    }
+
+    public void testFieldInjectionWithValidation() {
+        BeanValidation r = bind("{x:1,y:'2',z:'3'} }", BeanValidation.class);
```

Comment:
The json seems malformated due to the two trailing `}`. The test should fail from my PoV.

Actually all the json are invalid, the lib seems to be **really** lenient, it's not just a trailing `,` but a `{}` level issue here.